setup/configure_tpm_emulator/test.sh

```diff
@@ -48,16 +53,64 @@ User=tss
 [Install]
 WantedBy=multi-user.target
 _EOF"
+        fi
+
+        # for Fedora we use swtpm TPM emulator
+        if rlIsFedora; then
+            rlRun "yum -y install swtpm swtpm-tools tpm2-tss selinux-policy-devel tpm2-abrmd tpm2-tools"
+            # create swtpm unit file as it doesn't exist
+            rlRun "cat > /etc/systemd/system/swtpm.service <<_EOF
+[Unit]
+Description=swtpm TPM Software emulator
+
+[Service]
+Type=fork
+ExecStartPre=/usr/bin/mkdir -p /var/lib/tpm/swtpm
+ExecStartPre=/usr/bin/swtpm_setup --tpm-state /var/lib/tpm/swtpm --createek --decryption --create-ek-cert --create-platform-cert --lock-nvram --overwrite --display --tpm2
+ExecStart=/usr/bin/swtpm socket --tpmstate dir=/var/lib/tpm/swtpm --log level=4 --ctrl type=tcp,port=2322 --server type=tcp,port=2321 --flags startup-clear --tpm2
+
+[Install]
+WantedBy=multi-user.target
+_EOF"
+            # update tpm2-abrmd unit file
+            rlRun "cat > /etc/systemd/system/tpm2-abrmd.service <<_EOF
+[Unit]
+Description=TPM2 Access Broker and Resource Management Daemon
+# These settings are needed when using the device TCTI. If the
+# TCP mssim is used then the settings should be commented out.
+#After=dev-tpm0.device
+#Requires=dev-tpm0.device
+ConditionPathExistsGlob=
+
+[Service]
+Type=dbus
+BusName=com.intel.tss2.Tabrmd
+ExecStart=/usr/sbin/tpm2-abrmd --tcti=swtpm
+User=tss
+
+[Install]
+WantedBy=multi-user.target
+_EOF"
+            # now we need to build custom selinux module making swtpm_t a permissive domain
```

Comment:
Is this something we should report upstream?

Reply:
Maybe. Upstream swtpm is not using systemd unit file, probably on purpose due to many various ways how swtpm can be configured and started. I assume this might the reason why SELinux module doesn't have rules for transition from `init_t` to `swtpm_t` domain. This is one thing added by my module. However, even after adding these rules I have noticed some AVCs. Partially due to my configuration (accessing files in /var/lib/tmp), partially unrelated. I will try to gather more logs and report an issue upstream to find out.